### PR TITLE
Add a keepalive for Redis

### PR DESCRIPTION
**Pull request by Shiranuit**
## What does this PR do ?

Add a Keepalive mechanism that ping redis every 60s to keep the connection alive on Microsoft Azure

### CODE REVIEW

**Review by Aschen — APPROVED** [review]

File: lib/service/cache/redis.js [hunk 1]
```diff
@@ -93,6 +96,34 @@ class Redis extends Service {
...
   }
 
+  /**
+   * Setup a ping interval to keep the connection alive
+   * Every 60 seconds a ping is sent to Redis
+   */
+  _setupKeepAlive() {
+    this.client.on('ready', async () => {
+      await this._ping();
+      this.pingIntervalID = setInterval(this._ping.bind(this), 60000);
```
Comment (alexandrebouthinon) [inline lib/service/cache/redis.js:106]: I personnaly don't like hardcoded value in code. You could create a new configuration field to:
- enable/disable the ping mechanism
- configure the ping delay

**Review by alexandrebouthinon — CHANGES_REQUESTED** [review]

File: lib/service/cache/redis.js (hunk 1 above)
Comment (alexandrebouthinon) [reply lib/service/cache/redis.js:106]: This new configuration should be use later when IoRedis ping implementation will be ready

Comment (alexandrebouthinon) [reply lib/service/cache/redis.js:106]: And I think this should be disabled by default to avoid edge effects on other installation that do not require it

Comment (Shiranuit) [reply lib/service/cache/redis.js]: I did add a config that enables the Ping Keep Alive and sets its delay

**Review by alexandrebouthinon — APPROVED** [review]

**Review by alexandrebouthinon — CHANGES_REQUESTED** [review]
Missing default value in https://github.com/kuzzleio/kuzzle/blob/master/lib/config/default.config.js

**Comment by Shiranuit** [comment]
> Missing default value in https://github.com/kuzzleio/kuzzle/blob/master/lib/config/default.config.js

I could add a default value to 0, but since the config field is optional I dont know if it makes sense

**Review by alexandrebouthinon — APPROVED** [review]
You're right, It's ok for me

**Pull request merged**